yt/data_objects/construction_data_containers.py

```diff
@@ -185,13 +185,20 @@ def __init__(
                 removal="4.2",
             )
             method = style
+            issue_deprecation_warning(
+                "The 'mip' method value is a deprecated alias for 'max'. "
+                "Please use max directly.",
+                since="4.2",
+                removal="5.2",
```

Comment:
another important thing is that you should make sure that `stack_level` is correctly set so that users will see the actual line *they* typed in the warning rather than some random line internal to yt that they know nothing about.

Reply:
Okay, I updated since to 4.1 and removed removal.

Regarding `stack_level` I tried grep -rin "stack_level" from the top level but it didn't return anything. So then I looked at a few other deprecation warnings and it doesn't look like they include it either. So I guess this is a larger issue (?) For example, the deprecation warning for style instead of method in ProjectionPlot is what I copied from...

Reply:
My mistake, the argument is called `stacklevel`. I just tested it and here I think it should be set to `stacklevel=4` so we see the line where the YTProj object is created.
It would also be helpful to have a separate deprecation warning closer to users, directly in `AxisAlignedProjectionPlot.__init__`, and make sure we don't trigger both warnings at the same time (meaning, `AxisAlignedProjectionPlot` should also be able to replace `"mip"` with `"max"` so the `YTProj` object created internally doesn't need to emit a second warning)

Reply:
It doesn't look like stacklevel is being used anywhere so far:

```
(yt-fork-env) ryanjsfx@MB-145 yt-fork % grep -rin stacklevel .
./yt/_maintenance/deprecation.py:19:    msg: str, *, since: str, removal: Optional[str] = None, stacklevel: int = 3
./yt/_maintenance/deprecation.py:50:    warnings.warn(msg, VisibleDeprecationWarning, stacklevel=stacklevel)
```
but okay I set stacklevel=4 (will commit after I get the cookbook working).

Theoretically, should that also be true of the deprecation warning for the use of 'style' instead of 'method' keyword arguments for ProjectionPlot?

====
Regarding AxisAlignedProjectPlot.__init__ triggering the deprecation warning, should that also be true for the style vs method usage then? If so, maybe beyond the scope of this PR...

Reply:
> It doesn't look like stacklevel is being used anywhere so far:

this is indeed a problem we'll want to fix globally. I wasn't aware of this, thank you for calling it out !

> Theoretically, should that also be true of the deprecation warning for the use of 'style' instead of 'method' keyword arguments for ProjectionPlot?

yes !

> Regarding AxisAlignedProjectPlot.init triggering the deprecation warning, should that also be true for the style vs method usage then? If so, maybe beyond the scope of this PR.

Agreed. You don't need to fix these in your PR, but thanks a lot for raising this concern, I'll dedicate a PR to fixing this globally ! However my point about adding a depr to `AxisAlignedProjectionPlot` still stands :)

Reply:
Okay I moved the deprecation warning to AxisAlignedProjectionPlot (which will propagate here once I get pre-commit happy...)

Is `class YTProj` guaranteed to inherit from the axis-aligned version of ProjectionPlot though? If not, the deprecation won't trigger.

Reply:
> Okay I moved the deprecation warning to AxisAlignedProjectionPlot (which will propagate here once I get pre-commit happy...)

There's a slight misunderstanding. I meant that we want two deprecation warnings to exist, not move the one you already had.

> Is class YTProj guaranteed to inherit from the axis-aligned version of ProjectionPlot though? If not, the deprecation won't trigger.

So `YTProj` doesn't actually inherit anything from these classes, but I think I know what you mean, and indeed the deprecation should also be seen for `OffAxisProjectionPlot` so it's best to put it in the base `ProjectionPlot` class.

Reply:
Sooooo `class ProjectionPlot(NormalPlot)` only has a `__new__` method (no `__init__`) and it's not very clear to me if it takes in `method` (maybe as part of *args (?))

I put in the deprecation warning but doesn't look like it fits aesthetically at least 🤔 

Nerp it yields an UnboundLocalError method referenced before assignment

Reply:
Yeah this is delicate. Maybe just don't worry about it, I can do it myself once the rest of the PR is ready :)

Reply:
Yeah this is delicate. Maybe just don't worry about it, I can do it myself once the rest of the PR is ready :)